solana/program_v2/src/GatewayService.ts

```diff
@@ -0,0 +1,344 @@
+    }
+
+    return new Program<GatewayV2>(
+      idl,
+      GATEWAY_PROGRAM,
+      provider
+    ) as Program<GatewayV2>;
+  }
+
+  private constructor(
+    private _program: Program<GatewayV2>,
+    private _dataAccount: PublicKey,
+    private _cluster: ExtendedCluster = SOLANA_MAINNET,
+    private _wallet: Wallet = new NonSigningWallet(),
+    private _opts: ConfirmOptions = AnchorProvider.defaultOptions()
+  ) {}
+
+  static async createNetworkAddress(
+    authority: PublicKey
+  ): Promise<[PublicKey, number]> {
+    return findProgramAddress(authority);
+  }
+
+  getWallet(): Wallet {
+    return this._wallet;
+  }
+
+  getConnection(): Connection {
+    return this._program.provider.connection;
+  }
+
+  getConfrirmOptions(): ConfirmOptions {
+    return this._opts;
+  }
+
+  getIdl(): Idl {
+    return this._program.idl;
+  }
+
+  closeNetwork(
+    destination: PublicKey = this._wallet.publicKey,
+    authority: PublicKey = this._wallet.publicKey
+  ): GatewayServiceBuilder {
+    const instructionPromise = this._program.methods
+      .closeNetwork()
+      .accounts({
+        network: this._dataAccount,
+        systemProgram: anchor.web3.SystemProgram.programId,
+        destination,
+        authority,
+      })
+      .instruction();
+
+    return new GatewayServiceBuilder(this, {
+      instructionPromise,
+      didAccountSizeDeltaCallback: () => {
+        throw new Error('Dynamic Alloc not supported');
+      },
+      allowsDynamicAlloc: false,
+      authority,
+    });
+  }
+
+  createNetwork(
+    data: CreateNetworkData = {
+      authThreshold: 1,
+      passExpireTime: 16,
+      signerBump: 0,
+      fees: [],
+      authKeys: [{ flags: 1, key: this._wallet.publicKey }],
+    },
+    authority: PublicKey = this._wallet.publicKey
+  ): GatewayServiceBuilder {
+    // console.log("Creating with auth: " + authority.toBase58());
+
+    const instructionPromise = this._program.methods
+      .createNetwork({
+        authThreshold: data.authThreshold,
+        passExpireTime: new anchor.BN(data.passExpireTime),
+        fees: data.fees,
+        authKeys: data.authKeys,
+      })
+      .accounts({
+        network: this._dataAccount,
+        systemProgram: anchor.web3.SystemProgram.programId,
+        authority,
+      })
+      .instruction();
+
+    return new GatewayServiceBuilder(this, {
+      instructionPromise,
+      didAccountSizeDeltaCallback: () => {
+        throw new Error('Dynamic Alloc not supported');
+      },
+      // TODO: Implement this...
+      allowsDynamicAlloc: false,
+      authority,
+    });
+  }
+
+  updateNetwork(
+    data: UpdateNetworkData,
+    authority: PublicKey = this._wallet.publicKey
+  ): GatewayServiceBuilder {
+    const instructionPromise = this._program.methods
+      .updateNetwork({
+        authThreshold: data.authThreshold,
+        passExpireTime: new anchor.BN(data.passExpireTime),
+        // TODO?? Why do fees and authKeys have to be 'never' type??
```

Comment:
Let's not forget this...

Reply:
Doing @ts-ignore for now until we can get a better fix... it still works. I think it's a ts + anchor issue